### PR TITLE
persist-client: separate state of multiple shards in Consensus

### PR DESCRIPTION
**Pull request by aljoscha**
Before, one Consensus would be shared between all opened shards while the state stored in Consensus was only meant to be the state of a single shard.

This adds keys to the methods of the Consensus trait so that we can differentiate between different shards.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### CODE REVIEW

File: src/persist/src/location.rs [hunk 1]
```diff
@@ -949,21 +956,24 @@ pub mod tests {
...
                 .await,
             Ok(Ok(()))
         );
 
         // We can observe the a recent value on successful update.
-        assert_eq!(consensus.head(deadline).await, Ok(Some(new_state.clone())));
+        assert_eq!(
+            consensus.head(key, deadline).await,
+            Ok(Some(new_state.clone()))
+        );
```
Comment (danhhz) [inline src/persist/src/location.rs:976]: let's either add some test coverage for different keys to this or a TODO about it

**Review by danhhz — APPROVED** [review]
LGTM!

File: src/persist/src/location.rs (hunk 1 above)
Comment (ruchirK) [reply src/persist/src/location.rs:976]: leave it as a todo I can add more tests

**Review by ruchirK — APPROVED** [review]
LGTM

**Pull request merged**